File: src/main/java/com/gdosoftware/biblioteca/domain/interfaces/ISocioRepository.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package com.gdosoftware.biblioteca.domain.interfaces;

import com.gdosoftware.biblioteca.domain.modelo.Socio;
import java.util.List;

/**
 *
 * @author Dani
 */
public interface ISocioRepository {
    public Socio create(Socio libro);
    public Socio update(Long id, Socio libro);
    public Socio retrieve (Long id);
    public void deltete (Long id);
    public List<Socio> findAll();
}

File: domain/interfaces/iSocioRepository.go
package interfaces

import "github.com/gdosoftware/biblioteca/domain/modelo"

type ISocioRepository interface {
	Create(socio *modelo.Socio) (*modelo.Socio, error)
	Update(id string, socio *modelo.Socio) (*modelo.Socio, error)
	Retrieve(id string) (modelo.Socio, error)
	Delete(id string) error
	FindAll() ([]modelo.Socio, error)
}
